### PR TITLE
Enabled scope services to support Red Hat

### PR DESCRIPTION
**Pull request by tgoettlicher**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged ea81b5bc6ff6557200882289cb756708afe5d16a into 23f8d0bf3105e4507602b1ff020b7db0258181c3 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 7957469bdf0416ee256840395179c494cd0457a0 into 23f8d0bf3105e4507602b1ff020b7db0258181c3 )

File: plugins/inspect/services_inspector.rb [hunk 1]
```diff
@@ -42,6 +47,13 @@ def has_systemd(system)
     false
   end
 
+  def has_redhat_chkconfig(system)
+    system.run_command("chkconfig", "--version")
```
Comment (cornelius) [inline plugins/inspect/services_inspector.rb]: How does this check, if it has "redhat" chkconfig? It only tests for the existence of `chkconfig`, isn't it? How is this RedHat specific?

File: plugins/inspect/services_inspector.rb [hunk 2]
```diff
@@ -42,6 +47,13 @@ def has_systemd(system)
...
+    true
+  rescue Cheetah::ExecutionFailed
```
Comment (cornelius) [inline plugins/inspect/services_inspector.rb]: Checking for the existence of a tool, where it not being existant is a valid case, should not be done through exceptions. I would rather have a method in `System` which encapsulates that (something like `has_command?`) and return a boolean as result.

File: plugins/inspect/services_inspector.rb [hunk 3]
```diff
@@ -85,4 +97,29 @@ def inspect_sysvinit_services(system)
...
+    @summary = "Found #{output.lines.size} services."
+
+    services = output.lines.map do |line|
+      state = []
+      name, state[0], state[1], state[2], state[3], state[4], state[5], state[6] = line.split(/\s+/)
+      Service.new(name: name, state: state[runlevel.to_i].split(":")[1])
+    end
+
+    ServiceList.new(services.sort_by(&:name))
+  end
```
Comment (cornelius) [inline plugins/inspect/services_inspector.rb]: There is quite some duplication in this method with the `inspect_sysvinit_services` method. It might be better to explicitly code the difference into one method.

File: spec/unit/services_inspector_spec.rb [hunk 4]
```diff
@@ -71,11 +78,14 @@
...
 
-    it "returns data about SysVinit services when no systemd is present" do
+    it "returns data about SysVinit services on a suse system when no systemd is present" do
       system = double
       expect(system).to receive(:run_command).
         with("systemctl", "--version").
         and_raise(Cheetah::ExecutionFailed.new(nil, nil, nil, nil))
+      expect(system).to receive(:run_command).
+        with("chkconfig", "--version").
+        and_raise(Cheetah::ExecutionFailed.new(nil, nil, nil, nil))
```
Comment (cornelius) [inline spec/unit/services_inspector_spec.rb]: It's not very nice that this tests has to be adapted, when the code change actually is not related to what is tested by this test. Is there a better way?

Comment (tgoettlicher) [inline spec/unit/services_inspector_spec.rb]: That's because we need to find out if SUSE's or Red Hat's chkconfig is on the system. If we determined the OS by running the OS inspector first we could skip this step.

**Comment by tgoettlicher** [comment]
Squashed: https://github.com/SUSE/machinery/pull/525
